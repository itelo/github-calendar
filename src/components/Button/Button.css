.filter-button {
  border-radius: 3px;
  border-color: transparent;
  color: #586069;
  cursor: pointer;
  display: block;
  font-size: 14px;
  margin-bottom: 5px;
  overflow: hidden;
  padding: 8px 10px;
  padding-right: 48px;
  position: relative;
  text-decoration: none;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.filter-button:focus {
  outline: 0;
}

.filter-button:hover {
  background-color: #eaecef;
  text-decoration: none;
}

.filter-button.selected {
  background-color: #0366d6;
  color: #fff;
}
