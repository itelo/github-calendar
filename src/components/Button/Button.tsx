import * as React from "react";
import "./Button.css";

interface ButtonProps {
  isActive: boolean;
  onClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
  ["data-testid"]?: string;
}

const Button: React.FunctionComponent<ButtonProps> = props => {
  return (
    <button
      data-testid={props["data-testid"] ? props["data-testid"] : "button"}
      className={`filter-button ${props.isActive ? "selected" : ""}`}
      onClick={props.onClick}
    >
      {props.children}
    </button>
  );
};

export default Button;
