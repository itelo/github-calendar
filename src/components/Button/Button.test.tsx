import React from "react";
import { render, fireEvent } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import Button from "./Button";
import "jest-dom/extend-expect";

describe("<Button />", () => {
  test("test the click in button", () => {
    const handleClick = jest.fn();

    const { getByTestId } = render(
      <Button isActive={false} onClick={handleClick}>
        testing-the-button
      </Button>
    );

    const button = getByTestId("button");

    fireEvent.click(button);

    expect(handleClick).toBeCalledTimes(1);
  });

  test("test the inactive css in button", () => {
    const handleClick = jest.fn();

    const { getByTestId, container } = render(
      <Button isActive={false} onClick={handleClick}>
        testing-the-button
      </Button>
    );

    const button = getByTestId("button");

    expect(button).toHaveClass("filter-button");
    expect(button).not.toHaveClass("selected");
    expect(button).toContainHTML("testing-the-button");

    expect(container).toMatchSnapshot();
  });

  test("test the active css in button", () => {
    const handleClick = jest.fn();

    const { getByTestId, container } = render(
      <Button isActive={true} onClick={handleClick}>
        testing-the-button
      </Button>
    );

    const button = getByTestId("button");

    expect(button).toHaveClass("filter-button");
    expect(button).toHaveClass("selected");
    expect(button).toContainHTML("testing-the-button");

    expect(container).toMatchSnapshot();
  });
});
