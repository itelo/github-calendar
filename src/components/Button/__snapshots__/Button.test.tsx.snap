// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> test the active css in button 1`] = `
<div>
  <button
    class="filter-button selected"
    data-testid="button"
  >
    testing-the-button
  </button>
</div>
`;

exports[`<Button /> test the inactive css in button 1`] = `
<div>
  <button
    class="filter-button "
    data-testid="button"
  >
    testing-the-button
  </button>
</div>
`;
